### DISCUSSION TITLE
How to select the logging level of ffmpegio ?

### DISCUSSION THREAD

**Discussion by justinbacle** [python-ffmpegio/python-ffmpegio#11] [Q&A]
Hi all, 
I tried reducing the logging level of ffmpegio using the following 
`logging.getLogger("ffmpegio").setLevel(logging.WARNING)`
as I do it for other modules, but the path.py file of ffmpegio still continues to show loads of debug info on  ffmpegio.audio.read()
Is there a better/proper way to choose which log level to have ?

**Comment by tikuma-lsuhsc** [comment]
@justinbacle - Unfortunately, the logging aspect of this package has been coded very loosely (as I started building this package very early in my Python career). If you could print what the warning is, I can try to patch it up asap.

**Reply by justinbacle** [reply]
Thank you for the very quick answer :o
path.py on lines 167 and 186 seems responsible for all the logging I see

**Reply by tikuma-lsuhsc** [reply]
I see so the `logging.warning` calls... What's stinging me atm is that I never created a logger specific to `ffmpegio` (thus your `setLevel` doesn't work). 

I'll put revamping of the logging higher on my todo list, but unfortunately this will take more effort than a quick fix (and I'm currently swamped with work). So, for the moment, please run

`logging.basicConfig(level=logging.DEBUG)`

to lower the overall logging level. This should remove the logs (along with all other logs...)

**Reply by justinbacle** [reply]
Thank you, definitely okay to wait for a future update, if I find the time, I may try do it and create a PR, I thought I was missing something obvious.

**Reply by tikuma-lsuhsc** [reply]
I think I just realized what you're trying to do... to debug your own code while disabling debug message from dependent modules 😫 Sorry for the inconvenience

**Comment by tikuma-lsuhsc** [comment]
Took me a while, but the latest release, v0.9.0, switched from the root logger to `"ffmpegio"` logger.

**Reply by justinbacle** [reply]
Awesome, I'll try that in the coming weeks
